libs/vm/include/vm/state.hpp

```diff
@@ -149,10 +149,25 @@ class State : public IState
 
       // mark the variable as existed if we get a positive result back
       existed_ = (Status::OK == status);
+      if (!existed_)
+      {
+        Set(value);
+      }
     }
   }
 
-  ~State() override = default;
+  ~State() override
+  {
+    try
+    {
+      FlushIO();
+    }
+    catch (...)
```

Comment:
Think that it is worth catching the `std::exception` here and then added the message to the runtime error.

Reply:
Can do, I was originally thinking about it, but over the time I grew dislike of multiple `catch` blocks in name of lazyness and concise code (we will have to have `catch(...)` block anyway as ultimate fallback since we are in destructor).